### ISSUE TITLE
Visual redesign

### ISSUE THREAD

**Issue by pablobm** [thoughtbot/administrate#2360]
We have a few issues open about the current visual design of Administrate, tagged https://github.com/thoughtbot/administrate/labels/views-and-styles. These include:

- https://github.com/thoughtbot/administrate/issues/128
- https://github.com/thoughtbot/administrate/issues/794
- https://github.com/thoughtbot/administrate/issues/1367
- https://github.com/thoughtbot/administrate/issues/1783
- https://github.com/thoughtbot/administrate/issues/1856
- https://github.com/thoughtbot/administrate/issues/2247

This includes issues about accessibility and i18n, which should be borne in mind as part of this effort.

I think the main objectives here are the following:

- A responsive layout that works on a variety of devices and window sizes.
- An accessible user experience.
- Providing adequate i18n hooks.

Tackling this as a big redesign with a complete change of visuals would probably be a mistake, as we would start and never end. Instead we should take this in bite-sized, manageable pieces and solve it slowly. This may also encourage new contributors to participate.

As to what these pieces exactly should be... I'm not sure just yet :man_facepalming: Will update, but for now this serves as a central point for those conversations.

**Comment by Thrizian** [comment]
Are there any MR on this yet? I'd like to chip in my 50 cents. I would like to propose using some css grid here and there, Given this project aims for the future of admin interfaces, is it safe to say we would not be against that? I have customized the interface quite a bit locally. And can contribute my alterations if needed.

**Comment by Thrizian** [comment]
> #new-contributer here, Are there any MR on this yet? I'd like to chip in my 50 cents. I would like to propose using some css grid here and there, Given this project aims for the future of admin interfaces, is it safe to say we would not be against that? I have customized the interface quite a bit locally. And can contribute my alterations if needed.
> 
> The functionality i'd be proposing has 94% browser support according to caniuse, how much heart do we have for the 6 percents of browsers, where the support is: "unknown".

Mind not the colors, focus on the grid layout.
![Screen Shot 2024-10-08 at 08 55 38](https://github.com/user-attachments/assets/a9df0015-6713-45c7-b504-55cedb4b8cf3)
![Screen Shot 2024-10-08 at 08 56 46](https://github.com/user-attachments/assets/d7bec695-997d-4fdd-951c-0ee171bf0ca9)


The font is set to be: $base-font-size: clamp(16px, calc((1svw + 1svh) * .75), 3rem);
Which i find to work really well responsively. Sure on smaller screens the font gets smaller but you just set your own minimum at the clamp. and you're set to go.

**Comment by nickcharlton** [comment]
@Thrizian, would you be interested in opening a PR for this? I think it's a good start for bringing it in!

**Comment by Thrizian** [comment]
The screenshots were made of the hacks/monkey patches i've done in my own application to achieve that result, I'd have to take time to see if i can pull that apart from my application logic and blend it into how administrate works now. 

When time is in abundance, i'd love to be of help or think along. I really like the gist of Administrate @nickcharlton, so yes i'd like to. I wouldn't mind if we were to dedicate some time to this in a more broad sense where perhaps we get several people involved that put their heads together, I'm but a single human with opinions. Which might not represent that of the world. Based on the issuer of this ticket @pablobm 's view of things, I feel like he has a valuable and more broad vision over these aspects than i do.

When i find the time, i'll open a PR.

**Comment by pablobm** [comment]
Just wanted to second @nickcharlton and encourage @Thrizian to hack a bit on this, at his leisure, and open a draft PR where we can start looking at what's possible.

I don't really have a broad vision: the ticket was put together just after a couple of afternoons spent going through all open issues and putting together all that appeared to seem related to the visual design.
